background.tex

```diff
@@ -3,62 +3,63 @@
+Internal factors include heterogeneous capacity of worker nodes, and resource competition due to other tasks running on the same worker node. 
+External factors include resource competition due to co-hosted applications, input data skew, remote input or output source being too slow and faulty hardware~\cite{Chen2014,Dean2004}, and node mis-configuration~\cite{Dean2004}. 
+Garbage collection~\cite{Kyong2017,Ousterhout2017}, JVM positioning to cores~\cite{Kyong2017}, delays introduced while the tasks move from the scheduler to execution~\cite{Gittens2016}, disk I/O during shuffling, Java's just-in-time compilation~\cite{Ousterhout2017}, and output skew~\cite{Ousterhout2017} have also been found to introduce stragglers.
+In addition to these reasons, stragglers on Spark have been attributed to the overall performance of workers or competition between resources~\cite{Yang2016}.
+Garraghan et al.~\cite{Garraghan2016} reported high CPU utilization, disk utilization, unhandled I/O access requests and network package loss as the most frequent reasons for stragglers on Virtualized Cloud Data-centers.
 
-Tuning resource allocation and tuning parallelism such as breaking the workload into many small tasks that are dynamically scheduled at runtime \cite{Rosen2012}, slow Node-Threshold \cite{Dean2004}, speculative execution \cite{Dean2004}, sampling or data distribution estimation techniques, SkewTune to avoid data imbalance \cite{Kwon2012}, dynamic work rebalancing \cite{Schmidt2016}, blocked time analysis \cite{Ousterhout2015}, and intelligent scheduling \cite{AWE-WQ2014} are among a wide variety of approaches that are trying to detect and mitigate stragglers. 
+Tuning resource allocation and parallelism such as breaking the workload into many small tasks that are dynamically scheduled at runtime~\cite{Rosen2012}, slow Node-Threshold~\cite{Dean2004}, speculative execution~\cite{Dean2004}, sampling or data distribution estimation techniques, SkewTune to avoid data imbalance~\cite{Kwon2012}, dynamic work rebalancing~\cite{Schmidt2016}, blocked time analysis~\cite{Ousterhout2015}, and intelligent scheduling~\cite{AWE-WQ2014} are among a wide variety of approaches that are trying to detect and mitigate stragglers. 
 
-In the present study, we identify the root-cause of the straggler tasks in the context of analyzing large MD trajectories and provide solutions through which we can improve performance and scaling.
-Even though the proposed solutions to avoid stragglers are specifically applied to the analysis of MD trajectories in the \package{MDAnalysis} library, all of these principles and implementation techniques are potentially applicable to data analysis in other Python-based libraries.
+In the present study, we identify the cause of the straggler tasks in the context of analyzing large MD trajectories and provide solutions through which we can improve performance and scaling.
```

Comment:
Because we were looking for an initiating cause of either a condition or a causal chain that leads to an outcome or effect of interest. I think we need to keep it as root-cause which refers to the earliest, most basic, 'deepest', cause for a given behavior.

Reply:
I am not sure I agree with the distinction between cause and a root-cause in general. 

Regardless, my changes are suggestions. If you think something is not correct feel free to change it or ask for changes in the review and I will gladly make them!

Reply:
Of course, but we are discussing here to come up with the best final way. Please keep doing this and I will leave my comments as well and after @orbeckst comments, we will go with one way or the other.

Reply:
I would choose the slightly less emphatic "cause" as it puts slightly less burden of proof on us. We know that normal file access from multiple processes to the  trajectory on Lustre hurts performance. Can we consider this _the_ root-cause? Or is the true root-cause an issue with lustre/MPI interference? We have not really proven the latter. If we call our findings the root-cause then reviewers might ask us to dig into the network communication layer... 

@mkhoshle how would you summarize in 1-2 sentences the "root-cause"? 

However, I think we need to leave 

>  Even though the proposed solutions to avoid stragglers are specifically applied to the analysis of MD trajectories in the \package{MDAnalysis} library, all of these principles and implementation techniques are potentially applicable to data analysis in other Python-based libraries.

because we want to make clear that although we know that the results were obtained in the context of MDAnalysis, we think that the lessons are likely of wider applicability.